internal/tiltfile/probe.go

```diff
@@ -0,0 +1,181 @@
+)
+
+type probe struct {
+	spec *v1.Probe
+}
+
+func (p probe) String() string {
+	return p.spec.String()
+}
+
+func (p probe) Type() string {
+	return "probe"
+}
+
+func (p probe) Freeze() {}
+
+func (p probe) Truth() starlark.Bool {
+	return p.spec != nil
+}
+
+func (p probe) Hash() (uint32, error) {
+	return starlark.String(p.spec.String()).Hash()
+}
+
+var _ starlark.Value = probe{}
+
+func (s *tiltfileState) probe(_ *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var initialDelayVal, timeoutVal, periodVal, successThresholdVal, failureThresholdVal value.Int32Value
+	var exec execAction
+	var httpGet httpGetAction
+	var tcpSocket tcpSocketAction
+	err := s.unpackArgs(fn.Name(), args, kwargs,
+		"initial_delay_secs?", &initialDelayVal,
+		"timeout_secs?", &timeoutVal,
+		"period_secs?", &periodVal,
+		"success_threshold?", &successThresholdVal,
+		"failure_threshold?", &failureThresholdVal,
+		"exec?", &exec,
+		"http_get?", &httpGet,
+		"tcp_socket?", &tcpSocket,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	spec := &v1.Probe{
+		InitialDelaySeconds: int32(initialDelayVal),
+		TimeoutSeconds:      int32(timeoutVal),
+		PeriodSeconds:       int32(periodVal),
+		SuccessThreshold:    int32(successThresholdVal),
+		FailureThreshold:    int32(failureThresholdVal),
+		Handler: v1.Handler{
+			HTTPGet:   httpGet.action,
+			Exec:      exec.action,
+			TCPSocket: tcpSocket.action,
+		},
+	}
+
+	return probe{spec: spec}, nil
+}
+
+type execAction struct {
+	action *v1.ExecAction
+}
+
+func (e execAction) String() string {
+	return e.action.String()
+}
+
+func (e execAction) Type() string {
+	return "exec_action"
+}
+
+func (e execAction) Freeze() {}
+
+func (e execAction) Truth() starlark.Bool {
+	return e.action != nil
+}
+
+func (e execAction) Hash() (uint32, error) {
+	return starlark.String(e.action.String()).Hash()
+}
+
+func (s *tiltfileState) execAction(_ *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var command value.StringSequence
+	err := s.unpackArgs(fn.Name(), args, kwargs, "command", &command)
+	if err != nil {
+		return nil, err
+	}
+	spec := &v1.ExecAction{Command: []string(command)}
+	return execAction{action: spec}, nil
+}
+
+type httpGetAction struct {
+	action *v1.HTTPGetAction
+}
+
+func (h httpGetAction) String() string {
+	return h.action.String()
+}
+
+func (h httpGetAction) Type() string {
+	return "http_get_action"
+}
+
+func (h httpGetAction) Freeze() {}
+
+func (h httpGetAction) Truth() starlark.Bool {
+	return h.action != nil
+}
+
+func (h httpGetAction) Hash() (uint32, error) {
+	return starlark.String(h.action.String()).Hash()
+}
+
+func (s *tiltfileState) httpGetAction(_ *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var host, scheme, path starlark.String
+	var port value.Int32Value
```

Comment:
Ah no this was just copy-pasta palooza 😛 - the threshold/time values are explicitly `int32` while the port's got that weird `intstr` wrapper so takes in a plain `int` (int size doesn't _really_ matter here since port range validation happens way later anyway). Good eyes, I'll fix that up

Reply:
Cleaned up as part of the larger refactor in 9e0adfed (also fixed for `tcpSocketAction`)